broker-core/src/main/java/org/apache/qpid/server/security/group/FileGroupDatabase.java

```diff
@@ -104,17 +105,21 @@ public synchronized void setGroupFile(String groupFile) throws IOException
     @Override
     public synchronized void addUserToGroup(String user, String group)
     {
-        Set<String> users = _groupToUserMap.get(keySearch(_groupToUserMap.keySet(), group));
-        if (users == null)
+        Set<String> groupUsers = _groupToUserMap.get(keySearch(_groupToUserMap.keySet(), group));
+        if (groupUsers == null)
         {
             throw new IllegalArgumentException("Group "
                                                + group
                                                + " does not exist so could not add "
                                                + user
                                                + " to it");
         }
+        else if (groupUsers.contains(keySearch(_userToGroupMap.keySet(), user)))
+        {
+            throw new IllegalConfigurationException(String.format("Group member with name'%s' already exists", user));
+        }
 
-        users.add(keySearch(users, user));
+        groupUsers.add(keySearch(groupUsers, user));
```

Comment:
lt seems it would be safe to swap a call  `keySearch(groupUsers, user)` with a local variable introduced for `keySearch(_userToGroupMap.keySet(), user)`. Though if these to operations return different results, that would indicate an inconsistency and an exception would need to be thrown

Reply:
This call to keysearch is made explicitly not to add duplicate for a case sensitive provider. So replacing this will lead to undesriable results

Reply:
Could you please add an explicit check here that group user is the same as a user key

    final String groupKey = keySearch(groupUsers, user);
    if (!userKey.equals(groupKey))
    {
        throw new IllegalConfigurationException(String.format("Inconsistent data: user  key '%s' is not equal to a group key '%s'", userKey, groupKey));
    }